elide-core/src/main/java/com/yahoo/elide/security/executors/ActivePermissionExecutor.java

```diff
@@ -197,6 +202,59 @@ public ActivePermissionExecutor(final com.yahoo.elide.core.RequestScope requestS
     }
 
     /**
+     * Get permission filter on an entity.
+     *
+     * @param resourceClass Resource class
+     */
+    public FilterExpression getReadPermissionFilter(Class<?> resourceClass) {
+        if (requestScope.getSecurityMode() == SecurityMode.SECURITY_INACTIVE) {
+            return null; // Bypass
+        }
+        EntityDictionary dictionary = requestScope.getDictionary();
+        FilterExpressionVisitor fev = new FilterExpressionVisitor(dictionary, requestScope);
+        ParseTree classPermissions = dictionary.getPermissionsForClass(resourceClass, ReadPermission.class);
+        if (classPermissions != null) {
+            FilterExpression classPermission = fev.visit(classPermissions);
+            if (classPermission.isDirty()) {
+                classPermission = null;
+            }
+            FilterExpression fieldPermissions = getFieldPermissionFilter(resourceClass, fev);
+            if (fieldPermissions != null) {
+                return new OrFilterExpression(classPermission, fieldPermissions);
+            } else {
+                return classPermission;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    private FilterExpression getFieldPermissionFilter(Class<?> resourceClass,
+                                                      FilterExpressionVisitor fev) {
+        FilterExpression allFieldsExpression = null;
+        EntityDictionary entityDictionary = requestScope.getDictionary();
+        List<String> fields = entityDictionary.getAllFields(resourceClass);
+        for (String field : fields) {
+            ParseTree fieldPermissions =
+                    entityDictionary.getPermissionsForField(resourceClass, field, ReadPermission.class);
+            if (fieldPermissions == null) {
+                continue;
+            }
+            FilterExpression permissionFilter = fev.visit(fieldPermissions);
```

Comment:
This whole logic might be cleaner if we added visit methods to Expression, and instead of creating individual visitors on the permissions per field, we create a single composite Expression for the entire ReadPermission - and then visit that.